aws/ebs_test.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"testing"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestEBSVolume(t *testing.T, session *session.Session, name string) ec2.Volume {
+	svc := ec2.New(session)
+	volume, err := svc.CreateVolume(&ec2.CreateVolumeInput{
+		AvailabilityZone: awsgo.String("us-west-2a"),
+		Size:             awsgo.Int64(8),
+	})
+
+	if err != nil {
+		assert.Failf(t, "Could not create test EBS volume: %s", errors.WithStackTrace(err).Error())
+	}
+
+	err = svc.WaitUntilVolumeAvailable(&ec2.DescribeVolumesInput{
+		VolumeIds: []*string{awsgo.String(*volume.VolumeId)},
+	})
+
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	// Add test tag to the created instance
+	_, err = svc.CreateTags(&ec2.CreateTagsInput{
+		Resources: []*string{volume.VolumeId},
+		Tags: []*ec2.Tag{
+			{
+				Key:   awsgo.String("Name"),
+				Value: awsgo.String(name),
+			},
+		},
+	})
+
+	if err != nil {
+		assert.Failf(t, "Could not tag EBS volume: %s", errors.WithStackTrace(err).Error())
+	}
+
+	return *volume
+}
+
+func findEBSVolumesByNameTag(output *ec2.DescribeVolumesOutput, name string) []*string {
+	var volumeIds []*string
+	for _, volume := range output.Volumes {
+		// Retrive only IDs of instances with the unique test tag
+		for _, tag := range volume.Tags {
+			if *tag.Key == "Name" {
+				if *tag.Value == name {
```

Comment:
Instead of a second level of nesting, you could combine the two if-statements: `if *tag.Key == "Name" && *tag.Value == name`.

Reply:
Also, I wonder if there is a chance `tag.Key` or `tag.Value` could be a nil pointer? Might be safer to use `aws.StringValue`.